Make toggle

jonleesy opened the pull request.

jonleesy committed:
Commit

jonleesy committed:
Commit

jonleesy committed:
commit

jonleesy committed:
Merge branch 'main' into make-toggle

jonleesy committed:
commit

jonleesy committed:
Created Checkbox

jonleesy committed:
Delete toggle.css

jonleesy committed:
Update map.js

jonleesy committed:
Update map.js

kymckay commented on public/index.html:
```diff
@@ -33,6 +34,13 @@
+  
+
+  <!-- Rounded switch -->
+  <div class="toggle">
```
Think checkbox should go beside source dropdown

jonleesy replied:
Hahah. Guess we're not doing it on the map anymore?

kymckay committed:
Correct logical errors in colouring code

jonleesy committed:
nonb working commit

kymckay committed:
Respect toggle state whenever colours change

Relevant because they may be changed from various places (year change,
source change, toggle change)

kymckay committed:
Update map colouring when source changes

kymckay commented:
@jonleesy I fixed the code, but the start state is currently inverted (it starts checked, but should be unchecked as "Actual")

jonleesy committed:
Fix test error

jonleesy committed:
Merge branch 'main' into make-toggle

jonleesy committed:
Starts with actual

jonleesy committed:
remove error

jonleesy commented:
@kymckay should the toggle be "beautified" in this branched or should we just merged it first and do it as a feature?

kymckay commented:
@jonleesy Probably merge first, but I think it should be moved to beside the sources dropdown

jonleesy commented:
I'm actually working on it already. And I've managed to have it on the map

jonleesy committed:
Update index.html

jonleesy committed:
Create toggle.css

jonleesy committed:
Update toggle.css

jonleesy committed:
Add "Data"

jonleesy commented:
@kymckay would be great if you could review, then I could take screenshots from mains and draft the document

kymckay reviewed (approved):
Nice, looks decent.

I see what you mean about interacting with the map though, it does seem to be something on this branch causing it 🤔

jonleesy commented:
> Nice, looks decent.
> 
> I see what you mean about interacting with the map though, it does seem to be something on this branch causing it 🤔

I think I had a function that keeps constituencies highlighted upon click

kymckay commented:
>I think I had a function that keeps constituencies highlighted upon click

Nah it seems like the cursor is interactive a fixed distance above where it actually is (like constituencies become highlighted if you hover below them)

jonleesy committed:
Update index.html

kymckay commented:
I don't see anything in the code, I think it must be something about the HTML structure that mapbox doesn't like

kymckay commented:
Yeah confirmed, if you load the page and inspect the toggle and delete it suddenly the pointer starts working again. I think you need to put the toggle element outside of the `id="map"` div maybe

jonleesy commented:
<img width="418" alt="image" src="https://user-images.githubusercontent.com/65177718/112388985-e6a97300-8ceb-11eb-843c-629bf93e33f9.png">

tried it, it still does it

jonleesy commented:
Ill take that back. Somehow its better.

jonleesy commented:
@kymckay Its done. Place it on top instead. Not sure how the dropdown is configured, seemed pretty stubborn. I think I have given up

jonleesy committed:
toggle beside sources

jonleesy commented:
![image](https://user-images.githubusercontent.com/65177718/112390797-c0390700-8cee-11eb-8c34-7ab8810ebb63.png)
Looks like this now

kymckay commented:
@jonleesy Nice, I'll attempt to tweak the placement a bit then merge

kymckay commented:
You need to push changes though

jonleesy committed:
oops. Pushed

kymckay committed:
Improve affordance of map toggle

Clearly label the toggle as relating to the map

kymckay commented:
I couldn't be bothered trying to properly lay the toggle out so just made the labelling more clear that it's for the map to improve affordance 😅
